### DISCUSSION TITLE
Broadcasting a playlist consisting of identically encoded files, without transcoding

### DISCUSSION THREAD

**Discussion by omgiafs** [savonet/liquidsoap#3735] [Q&A]
Hello!
I want to ask a simple question.

I have a playlist that contains files encoded with the same parameters: mp3 CBR 320 kbps.
Is it possible to launch a radio from this playlist without using transcoding or using it minimally, only on transitions between tracks?

If I use something like

```
myplaylist = playlist("/playlists/playlist_local.m3u")
security = single("/content/default.mp3")

# Start building the feed with music
radio = myplaylist
radio = fallback(track_sensitive = false, [radio, security])

# Stream it out
output.icecast(format="audio/mpeg", %ffmpeg(format="mp3", %audio.copy),
   fallible=true,
   mount = "/radio",
   host = "icecast",
   port = 9001,
   password = "hackme")
```

, the client's playback stops at the end of the track. When the client reconnects, playback resumes, but until the end of the track.

My position: I find it unnecessary to waste CPU time constantly transcoding a source that is guaranteed to contain identically encoded information. I accept that track transitions (crossfade or silent, which are not included in the example) can and should be transcoded. But I find it completely wrong to transcode for the entire duration of a track, given also that it is a lossy source.
I can use any type of destination, it doesn’t matter.

What functions can be used to minimize transcoding in this case?

**Comment by toots** [comment]
Hi,

Thanks for your comment. I'm not sure about the behavior of the script you described above -- it should be streaming continuously.

Regarding encoding only around transitions, I am not sure that this is technically possible in general. It might work for simple formats such as mp3 but for more tricky ones I am not sure that would be able to keep the internal codec consistency when decoding and re-encoding data only on a portion of the stream.

We could also allow only for mp3 but this seems like a lot of work for us so, due to limited developer time this might not be our highest priority. However, if you're interested to work on it, I would happy to provide guidance.
